### PR TITLE
Globus & Gantry monitor services + API endpoints

### PR DESCRIPTION
**Pull request by max-zilla**
As discussed in #79, there are 2 components in /scripts/globusmonitor:

**gantry_monitor_service.py**
This has a service component which will monitor a directory for files that are more than 15 mins old (or whatever config time is). It will add those files to a PendingTransfers queue, organized by dataset (dataset = "sensorName timestamp"). There is also a lightweight API component that allows someone on that network to manually submit files to the PendingTransfers queue - MAC script can do this to avoid waiting for the 15-min crawler to catch the file.

Every X minutes a Globus transfer is started with everything in PendingTransfers, and a message is sent to the globus_monitor_service API on the NCSA side with the Globus task ID. The task is added to a log listing time, component files, etc.

Every Y minutes the NCSA API is queried about the tasks in this log. When that API reports the task as complete and the file was successfully processed into the Clowder pipeline, the gantry-side original files are moved to a folder where they can be queued for deletion. 

**globus_monitor_service.py**
This has a service component that will query Globus every N minutes regarding the transfer status of the globus tasks it is monitoring. There is also an API component that accepts new tasks to monitor, as well as GET task details and remove tasks from monitoring.

When the API receives a new task (typically from the gantry_monitor_service), it will continually check the status of the transfer until it is either SUCCEEDED or FAILED. On success, this service will submit all files included in that task to Clowder for processing, creating the dataset if necessary, and writing to a log file. It maintains a map of datasetName -> clowderDatasetUUID, so when a dataset named "sensorName timestampXYZ" is created, any other files that arrive with that same dataset name will go in the same Clowder dataset. The file is submitted to clowder using a local file path, so the file is not moved out of the Globus landing directory even when processed.

**notes**
Both files have a companion config file. Generally you can specify IP and port numbers, globus user credentials and endpoint IDs (I determined them from [here](https://www.globus.org/app/endpoints)), various wait times between checks, and where to write log files/globus transfers.

**todos/questions**
- **I think we should have admin emails that can be listed in the config file**, who get an email if certain errors are encountered (e.g. gantry can't contact globus, NCSA API not responding, etc)
- want to dockerize both of these - both have some python dependencies (like requests) that need to be available in the environment
- Need some help thinking through **how best to simulate whole pipeline**. I have access to Roger and CampusCluster now, but not sure best practice for running the scripts there (installing python libs) or via docker. generally, i planned to create a /gantry folder on my Roger space and run the gantry monitor, then run the globus monitor on my laptop, and move files to the Roger /gantry folder so they're passed along to globus. 
- There's **still hardening to be done** - once this goes, it's meant to go 24/7 more or less. The pending/active/completed tasks and the dataset->UUID map are continually written to log files when updated with a single backup, so data should generally be recoverable on a crash. However most of the try/catch blocks for errors are fairly simple and I don't do a whole lot when API requests return 400s/404s beyond printing something out to the console (which **should really be a separate log file itself** now that I think about it).

But I thought I'd go ahead and make this pull request now to get some of these thoughts out there.

### CODE REVIEW

**Comment by max-zilla** [comment]
Just added the external logfile functionality.

**Comment by dlebauer** [comment]
@jdmaloney can you please review / provide any feedback on Max's pull request?

File: scripts/gantrymonitor/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    && pip install globusonline-transfer-api-client \
+    && pip install python-logstash \
+    && mkdir /home/gantry \
+    && chown -R data-mover /home/gantry \
+    && chown -R data-mover /var/log
+
+COPY gantry_monitor_service.py /home/gantry/gantry_monitor_service.py
+COPY config_default.json /home/gantry/config_default.json
+
+ENV MONITOR_API_PORT 5455
```
Comment (robkooper) [inline scripts/gantrymonitor/Dockerfile]: Can you add the volume command so you make explicit what volumes can be mounted.

**Comment by robkooper** [comment]
it is running so let's merge this. next pull requests should be smaller though.

**Comment by max-zilla** [comment]
@robkooper what would be the goal of squashing commits? just so there aren't 200 listed? I didn't really think about it as I was pushing updates and fixes without merging.

**Comment by robkooper** [comment]
The thinking is for you to be able to commit as much as you want, but when you are ready to publish you can combine commits together so you have a nice set of commits, this is especially nice in this case where you have 200 commits.

**Pull request merged**